main/src/cgeo/geocaching/speech/TextFactory.java

```diff
@@ -100,16 +100,21 @@ private static String getQuantityString(@PluralsRes final int resourceId, final
     }
 
     private static String getDirection(final Geopoint position, final Geopoint target, final float direction) {
-        final int bearing = (int) position.bearingTo(target);
-        final int degrees = (int) AngleUtils.normalize(bearing - direction);
+        int hours;
 
-        int hours = (degrees + 15) / 30;
-        if (hours == 0) {
+        if (position.distanceTo(target) < 1e-4) {
+            // By convention, distance smaller than ten centimers will be represented as 12 o'clock.
             hours = 12;
+        } else {
+            final int bearing = (int) position.bearingTo(target);
+            final int degrees = (int) AngleUtils.normalize(bearing - direction);
+
+            hours = (degrees + 15) / 30;
```

Comment:
I haven't touch this part except put in into braces. Feel free to open another issue if you think some things should be cleared up.